### PR TITLE
build(aio): fail build if more than h1 is found in a doc

### LINKED ISSUE #16193: Ban H1 elements (or single #) in document content
Some documents like [animation.md - see the "Contents" header](https://github.com/angular/angular/blob/800b1b060edfc05fde2172121a163a2e1c868ae2/aio/content/guide/animations.md) contain H1 headers which mess up UI and in the future also TOC generation.

It would be great to catch these errors in dgeni and fail a build if such errors are introduced.

### PR DESCRIPTION
**Pull request by petebacondarwin**
Closes #16193

### CODE REVIEW

File: packages/common/src/directives/ng_if.ts [hunk 1]
```diff
@@ -32,15 +32,15 @@ import {Directive, EmbeddedViewRef, Input, TemplateRef, ViewContainerRef} from '
...
  * {@example common/ngIf/ts/module.ts region='NgIfElse'}
  *
- * # Using non-inlined `then` template
+ * ## Using non-inlined `then` template
  *
  * Usually the `then` template is the inlined template of the `ngIf`, but it can be changed using
  * a binding (just like `else`). Because `then` and `else` are bindings, the template references can
  * change at runtime as shown in this example.
- *
+ *ß
```
Comment (gkalpak) [inline packages/common/src/directives/ng_if.ts]: ??

File: aio/tools/transforms/post-process-package/processors/post-process-html.js [hunk 2]
```diff
@@ -28,19 +28,20 @@ module.exports = function postProcessHtml(log, createDocMessage) {
...
 
+      let vFile;
+
       docs
         .filter(doc => this.docTypes.indexOf(doc.docType) !== -1)
         .forEach(doc => {
-          const vFile = engine.processSync(doc.renderedContent);
-          vFile.messages.forEach(m => {
-            const message = createDocMessage(m.message, doc);
-            if (m.fatal) {
```
Comment (gkalpak) [inline aio/tools/transforms/post-process-package/processors/post-process-html.js]: What does this mean for `fatal` errors? Are there any? Are we ignoring them (just logging)?

File: aio/tools/transforms/angular-base-package/post-processors/h1-checker.spec.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    const doc = {
+      docType: 'a',
+      renderedContent: `
+        <h1>Heading 1</h2>
+        <h2>Heading 2</h2>
+        <h1>Heading 1a</h1>
+    `
+    };
+    expect(() => processor.$process([doc])).toThrowError(createDocMessage('More than one h1 found [<h1>Heading 1, <h1>Heading 1a</h1>]', doc));
+  });
```
Comment (gkalpak) [inline aio/tools/transforms/angular-base-package/post-processors/h1-checker.spec.js]: What happens with 0 or 1 h1s? Where's the test 😛

**Review by gkalpak — CHANGES_REQUESTED** [review]

File: aio/tools/transforms/angular-base-package/post-processors/h1-checker.spec.js (hunk 3 above)
Comment (petebacondarwin) [inline aio/tools/transforms/angular-base-package/post-processors/h1-checker.spec.js]: OK OK

File: packages/common/src/directives/ng_if.ts (hunk 1 above)
Comment (petebacondarwin) [inline packages/common/src/directives/ng_if.ts]: weird!!

File: aio/tools/transforms/post-process-package/processors/post-process-html.js (hunk 2 above)
Comment (petebacondarwin) [inline aio/tools/transforms/post-process-package/processors/post-process-html.js]: So if you use `vFile.fatal()` to create a fatal error message, it immediately throws. So this code was never being used called. It is possible to create a message and manually set its fatal property, in which case we would only log it, along with warning messages.

**Comment by petebacondarwin** [comment]
@gkalpak PTAL

File: aio/tools/transforms/angular-base-package/post-processors/h1-checker.spec.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    };
+    expect(() => processor.$process([doc])).not.toThrow();
+  });
+
+  it('should not complain if there are no h1s in a document', () => {
+    const doc = {
+      docType: 'a',
+      renderedContent: `
+        <h2>Heading 2</h2>
+    `
```
Comment (gkalpak) [inline aio/tools/transforms/angular-base-package/post-processors/h1-checker.spec.js]: Nit: Uncommon indentation of closing `` ` ``.

**Review by gkalpak — APPROVED** [review]

File: aio/tools/transforms/angular-base-package/post-processors/h1-checker.spec.js (hunk 4 above)
Comment (petebacondarwin) [inline aio/tools/transforms/angular-base-package/post-processors/h1-checker.spec.js]: Maybe but it appears elsewhere in the transforms stuff, so...

**Review by IgorMinar — APPROVED** [review]

**Pull request merged**